mindsdb/integrations/libs/response.py

```diff
@@ -20,8 +24,11 @@ def type(self):
 
     def to_json(self):
         try:
-            data = self.data_frame.to_json(orient="split", index=False)
-        except Exception:
+            data = None
+            if self.data_frame is not None:
+                data = self.data_frame.to_json(orient="split", index=False, date_format="iso")
```

Comment:
Is better to use pickle as data serialization. It is faster and keeps data types
pickle5 module can allow to work with the same pickle protocol for different python version (works from py3.5)

Reply:
Agree, but it will be as a part of refactoring